contracts/potlock/src/potlock_interactions.rs

```diff
@@ -0,0 +1,78 @@
+use crate::potlock_storage::{self, Pot, Project};
+use crate::potlock_storage::{PotlockId, ProjectId};
+
+multiversx_sc::imports!();
+multiversx_sc::derive_imports!();
+
+#[multiversx_sc::module]
+pub trait PotlockInteractions:
+    potlock_storage::PotlockStorage + multiversx_sc_modules::only_admin::OnlyAdminModule
+{
+    #[payable("*")]
+    #[endpoint(addPot)]
+    fn add_pot(&self, name: ManagedBuffer, description: ManagedBuffer) {
+        let payment_for_adding_pot = self.call_value().single_esdt();
+        require!(
+            self.fee_token_identifier().get() == payment_for_adding_pot.token_identifier,
```

Comment:
Only one payment token for all pots? And an ESDT? So I guess it will be WEGLD?

Reply:
This is the token paid by the pot initiator (from the specs: _AddPot@potDescription - it sets the description and the name of the pot, the call requires a FEE in a defined tokenID._)

The fee token can be set to any token, probably WEGLD.
Another contract can be deployed for xExchange and the fee will probably be MEX.